autogpt_platform/backend/schema.prisma

```diff
@@ -19,45 +19,148 @@ model User {
   updatedAt DateTime @updatedAt
 
   // Relations
-  AgentGraphs                  AgentGraph[]
+  AgentGraphs                  Agent[]
   AgentGraphExecutions         AgentGraphExecution[]
   AgentGraphExecutionSchedules AgentGraphExecutionSchedule[]
   AnalyticsDetails             AnalyticsDetails[]
   AnalyticsMetrics             AnalyticsMetrics[]
   UserBlockCredit              UserBlockCredit[]
+  ConfiguredAgents             ConfiguredAgent[]
+  UserAgents                   UserAgents[]
+
+  // User Group relations
+  groups                 UserGroup[]              @relation("GroupUsers")
+  adminGroups            UserGroup[]              @relation("GroupAdminUsers")
+  UserAccounting         UserAccounting[]
+  UserSubscription       UserSubscription[]
+  Profile                Profile[]
+  StoreListing           StoreListing[]
+  StoreListingSubmission StoreListingSubmission[]
 
   @@index([id])
   @@index([email])
 }
 
+model UserGroup {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  name         String
+  description  String
+  groupIconUrl String?
+
+  users      User[] @relation("GroupUsers")
+  adminUsers User[] @relation("GroupAdminUsers")
+
+  agents       Agent[]
+  Profile      Profile[]
+  StoreListing StoreListing[]
+
+  @@index([name])
+}
+
 // This model describes the Agent Graph/Flow (Multi Agent System).
-model AgentGraph {
+model Agent {
   id        String    @default(uuid())
   version   Int       @default(1)
   createdAt DateTime  @default(now())
   updatedAt DateTime? @updatedAt
 
   name        String?
   description String?
-  isActive    Boolean @default(true)
-  isTemplate  Boolean @default(false)
 
   // Link to User model
-  userId String
-  user   User   @relation(fields: [userId], references: [id], onDelete: Cascade)
+  createdByUserId String
+  createdByUser   User   @relation(fields: [createdByUserId], references: [id], onDelete: Cascade)
 
-  AgentNodes                  AgentNode[]
-  AgentGraphExecution         AgentGraphExecution[]
-  AgentGraphExecutionSchedule AgentGraphExecutionSchedule[]
+  isGroupAgent Boolean    @default(false)
```

Comment:
This seems redundant with `groupId`: if `groupId` is set, it's a group-owned agent.